analyzers/src/SonarAnalyzer.Common/Rules/UseWhereBeforeOrderByBase.cs

```diff
@@ -38,14 +38,11 @@ public abstract class UseWhereBeforeOrderByBase<TSyntaxKind, TInvocation> : Sona
                 && Language.Syntax.TryGetOperands(invocation, out var left, out var right)
                 && LeftHasCorrectName(left, out var orderByMethodDescription)
                 && MethodIsLinqExtension(left, c.SemanticModel)
-                && MethodIsLinqExtension(right, c.SemanticModel))
+                && MethodIsLinqExtension(right, c.SemanticModel)
+                && Language.Syntax.NodeIdentifier(right) is { } rightIdentifier
+                && Language.Syntax.NodeIdentifier(left) is { } leftIdentifier)
```

Comment:
This is a change in logic. It suppresses issues where the leftIdentifier is null. Before, it was raised with an invalid/ignored secondary location, and now it is filtered out.

Reply:
Okay. But this change needs peach validation, then.

Reply:
Yes, good point!